### ISSUE TITLE
Enhancement: [no-confusing-void-expression] ignoreVoidReturningFunctions should suggest adding void annotation

### ISSUE THREAD

**Issue by phaux** [typescript-eslint/typescript-eslint#10290]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://deploy-preview-10067--typescript-eslint.netlify.app/rules/no-confusing-void-expression#ignorevoidreturningfunctions

### Description

When given code:

```ts
function foo() {
  if (cond) return console.log()
}
```

The error says to move the `console.log` before `return` and autofix does it automatically, but when `ignoreVoidReturningFunctions` is enabled a simpler fix would be to simply add the void annotation to the function return type.

I think it should mention this possibility in every error message and autofix should do whatever requires less changes. For example for arrow functions it could preserve the current behavior and fix by adding braces.

### Fail

```TypeScript
// no changes
```

### Pass

```TypeScript
// no changes
```

### Additional Info

Continued from #8538

**Comment by JoshuaKGoldberg** [comment]
Huh, I'm surprised nobody's commented on this till now. Sorry about that!

...though I'm -1 personally on trying to be clever with the fix. In cases like the example code, yes, a `: void` on the function would make sense. But what if it's a big function with an inferred type that happens to be a union including `| undefined`? What if it's also a recursive function?

We don't generally try to be smart and choose between better options in fixes, because it often ends up being a difficult trail of _"well, in this case, it's 51% better to switch to..."_.

cc @typescript-eslint/triage-team - thoughts?

**Comment by JoshuaKGoldberg** [comment]
Huh, I'm surprised nobody's commented on this till now. Sorry about that!

...though I'm -1 personally on trying to be clever with the fix. In cases like the example code, yes, a `: void` on the function would make sense. But what if it's a big function with an inferred type that happens to be a union including `| undefined`? What if it's also a recursive function?

We don't generally try to be smart and choose between better options in fixes, because it often ends up being a difficult trail of _"well, in this case, it's 51% better to switch to..."_.

@phaux I'm tempted to close this out, but want to hear from you first. Am I horribly misinterpreting / missing something?

**Comment by phaux** [comment]
@JoshuaKGoldberg I was interested in adding this fix but since #10289 is not going to be changed then I personally will not probably use this option so I guess it can be closed.
